2021/day15/day15.java

```diff
@@ -0,0 +1,146 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.PriorityQueue;
+import java.util.Scanner;
+import java.lang.Integer;
+
+public class day15 {
+    public static class Position implements Comparable<Position> {
+        int x;
+        int y;
+        int risk;
+
+        Position(int vertical, int horizontal, int currentRisk) {
+            x = vertical;
+            y = horizontal;
+            risk = currentRisk;
+        }
+
+        @Override
+        public int compareTo(Position other) {
+            return Integer.compare(this.risk, other.risk);
+        }
+    }
+
+    public static ArrayList<ArrayList<Integer>> readInput(String filename) {
+        ArrayList<ArrayList<Integer>> input = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                ArrayList<Integer> line = new ArrayList<>();
+                sc.useDelimiter("");
```

Comment:
Is this necessary? Isn't `""` the default delimiter?

Reply:
It is necessary. Using ```""``` allows for spliting input after every single character instead of after whole number (whole line).